Sources/Brave/Frontend/Sync/BraveCore/BraveSyncAPIExtensions.swift

```diff
@@ -60,10 +66,15 @@ extension BraveSyncAPI {
     deleteDevice(deviceGuid)
   }
 
-  func leaveSyncGroup() {
-    // Remove all observers before leaving the sync chain
-    removeAllObservers()
-
+  /// Method for leaving sync chain
+  /// Removing Observers, clearing local preferences and calling reset chain on brave-core side
+  /// - Parameter includeObservers: Parameter that decides if observers should be removed
+  func leaveSyncGroup(includeObservers: Bool = true) {
```

Comment:
`includeObservers` doesn't really explain what the arg is for in the context of the method name. One of the  following would be better:

```swift
func leaveSyncGroup(removingObservers: Bool = true) { ... }
func leaveSyncGroup(preservingObservers: Bool = false) { ... }
func leaveSyncGroup(preservingObservers: Bool = true) { ... } // And flipping the bool check below
```